website/docs/r/data_factory_data_flow.html.markdown

```diff
@@ -0,0 +1,215 @@
+  }
+
+  encoding = "UTF-8"
+}
+
+resource "azurerm_data_factory_dataset_json" "example2" {
+  name                = "dataset2"
+  resource_group_name = azurerm_resource_group.example.name
+  data_factory_name   = azurerm_data_factory.example.name
+  linked_service_name = azurerm_data_factory_linked_custom_service.example.name
+
+  azure_blob_storage_location {
+    container = "container"
+    path      = "foo/bar/"
+    filename  = "bar.txt"
+  }
+
+  encoding = "UTF-8"
+}
+
+resource "azurerm_data_factory_data_flow" "example" {
+  name            = "example"
+  data_factory_id = azurerm_data_factory.example.id
+
+  source {
+    name = "source1"
+
+    dataset {
+      name = azurerm_data_factory_dataset_json.example1.name
+    }
+  }
+
+  sink {
+    name = "sink1"
+
+    dataset {
+      name = azurerm_data_factory_dataset_json.example2.name
+    }
+  }
+
+  script = <<EOT
+source(
+  allowSchemaDrift: true, 
+  validateSchema: false, 
+  limit: 100, 
+  ignoreNoFilesFound: false, 
+  documentForm: 'documentPerLine') ~> source1 
+source1 sink(
+  allowSchemaDrift: true, 
+  validateSchema: false, 
+  skipDuplicateMapInputs: true, 
+  skipDuplicateMapOutputs: true) ~> sink1
+EOT
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) Specifies the name of the Data Factory Data Flow. Changing this forces a new resource to be created.
+
+* `data_factory_id` - (Required) The ID of Data Factory in which to associate the Data Flow with. Changing this forces a new resource.
+
+* `script` - (Required) The script for the Data Factory Data Flow.
+
+* `source` - (Required) One or more `source` blocks as defined below.
+
+* `sink` - (Required) One or more `sink` blocks as defined below.
+
+* `annotations` - (Optional) List of tags that can be used for describing the Data Factory Data Flow.
+
+* `description` - (Optional) The description for the Data Factory Data Flow.
+
+* `folder` - (Optional) The folder that this Data Flow is in. If not specified, the Data Flow will appear at the root level.
+
+* `transformation` - (Optional) One or more `transformation` blocks as defined below.
+
+---
+
+A `source` block supports the following:
+
+* `name` - (Required) The name for the Data Flow Source.
+
+* `description` - (Optional) The description for the Data Flow Source.
+
+* `dataset` - (Optional) A `dataset` block as defined below.
+
+* `linked_service` - (Optional) A `linked_service` block as defined below.
+
+* `schema_linked_service` - (Optional) A `schema_linked_service` block as defined below.
+
+---
+
+A `sink` block supports the following:
+
+* `name` - (Required) The name for the Data Flow Source.
+
+* `description` - (Optional) The description for the Data Flow Source.
+
+* `dataset` - (Optional) A `dataset` block as defined below.
+
+* `linked_service` - (Optional) A `linked_service` block as defined below.
+
+* `schema_linked_service` - (Optional) A `schema_linked_service` block as defined below.
```

Comment:
what are the differences between these two?

Reply:
as above